File: Spring/task3.Blackjack/task3.Blackjack/Dealer.cs
using System;
using System.Collections.Generic;


namespace Task3.Blackjack
{
    public class Dealer
    {
        private List<Card> cards;

        public int SumCards { get; private set; }

        public Card FirstCard { get; private set; }

        public Dealer()
        {
            cards = new List<Card>();
            SumCards = 0;
        }

        private void AddCard(Decks decks)
        {
            Card card = decks.Pop();
            cards.Add(card);
            SumCards += card.GetValueOfCard();
        }

        private void AddFirstCard(Decks decks)
        {
            Card card = decks.Pop();
            cards.Add(card);
            SumCards += card.GetValueOfCard();
            FirstCard = card;
        }

        public void GiveCard(APlayer player, Decks decks)
        {
            Card card = decks.Pop();
            player.AddCard(card);
        }

        public void Dealing(List<APlayer> players, Decks decks)
        {
            for(int i = 0; i < players.Count; i++)
            {
                GiveCard(players[i], decks);
            }
            AddFirstCard(decks);

            for (int i = 0; i < players.Count; i++)
            {
                GiveCard(players[i], decks);
            }
            AddCard(decks);
        }

        public void Print()
        {
            Console.Write("Dealer: {0} cards: ", cards.Count);
            for (int i = 0; i < cards.Count - 1; i++)
            {
                cards[i].Print();
                Console.Write(", ");
            }
            cards[cards.Count - 1].Print();
        }

        public void TakeCards(Decks decks)
        {
            while(SumCards < 17)
            {
                AddCard(decks);
            }
        }

        public void ResetCards()
        {
            cards.Clear();
            SumCards = 0;
        }
    }
}

File: Spring/task3/src/Dealer.java
import java.util.LinkedList;

public class Dealer extends LinkedList<Card>
{

    public Card firstCard;

    public Card getFirstCard() {
        return firstCard;
    }

    public int score;

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        score = score;
    }

    public Dealer()
    {
        score = 0;
    }

    private void AddCard(Deck deck)
    {
        Card card = deck.Pop();
        this.add(card);
        score += card.getScore();
    }

    private void AddFirstCard(Deck decks)
    {
        Card card = decks.Pop();
        this.add(card);
        score += card.getScore();
        firstCard = card;
    }

    public void GiveCard(Player player, Deck decks)
    {
        Card card = decks.Pop();
        player.AddCard(card);
    }

    public void Dealing(LinkedList<Player> players, Deck decks)
    {
        for(int i = 0; i < players.size(); i++)
        {
            GiveCard(players.get(i), decks);
        }
        AddFirstCard(decks);

        for (int i = 0; i < players.size(); i++)
        {
            GiveCard(players.get(i), decks);
        }
        AddCard(decks);
    }

    public void PrintInfo()
    {
        System.out.println("Dealer " + this.size() + " cards ");
        for (int i = 0; i < this.size() - 1; i++)
        {
            System.out.print(this.get(i).toString() + ", ");
        }
        System.out.println(this.get(this.size() - 1).toString());
    }

    public void TakeCards(Deck decks)
    {
        while(score < 17)
        {
            AddCard(decks);
        }
    }

    public void ResetCards()
    {
        this.clear();
        score = 0;
    }
}
